src/components/input/input.tsx

```diff
@@ -16,11 +17,10 @@ export interface InputProps {
 		| 'none'
 		| 'numeric'
 		| 'decimal';
-	shrinkLabel?: boolean;
 	onInputChange?: Function;
 	startAdornment?: JSX.Element;
 	endAdornment?: JSX.Element;
-	isValueValid?(value: string): boolean;
+	isValueValid?(value: string): boolean | Promise<boolean>;
```

Comment:
```suggestion
	isValueValid?(value: string): Promise<boolean>;
```
to keep the same type